rpc/blockchain.go

```diff
@@ -198,6 +225,16 @@ func (s *PublicBlockchainService) GetBlockByNumber(
 func (s *PublicBlockchainService) GetBlockByHash(
 	ctx context.Context, blockHash common.Hash, opts interface{},
 ) (response StructuredResponse, err error) {
+
+	if s.limiter != nil {
```

Comment:
Can we refactor this into a new function?

Reply:
good point!! This urgent task about OOM makes me temporarily forget an important principle in software engineering - Do not repeat yourself. Thanks for your reminder!!